test/jsx/indent.civet

```diff
@@ -1,6 +1,18 @@
 {testCase, throws} from ../helper.civet
 
 describe "Indentation-based JSX", ->
+  testCase """
+    no children
+    ---
+    return
+      <div>
+    console.log 'unreachable'
+    ---
+    return
+      <div />
```

Comment:
We should wrap nested return expressions in parens because JS/TS treats `return\n` as a statement end.

Don't let this block the feature though, it's likely a more general problem in the parser currently.

Reply:
Ah, good point. I noticed CoffeeScript has an explicit "RETURN INDENT ... DEDENT" rule. Maybe we want to add one to Civet. (This would also improve the indentation of the added closing tags.) Then we could easily add parens in that rule.

Reply:
I sort of already have this in `MaybeNestedExpression` (probably needs a better name). Essentially if there is a `&EOS ExtendedExpression` we should make sure to wrap in parens.